druid/src/widget/widget.rs

```diff
@@ -222,6 +223,38 @@ pub trait Widget<T> {
             ..Default::default()
         }
     }
+
+    /// Computes max intrinsic/preferred dimension of a widget on the provided axis.
+    ///
+    /// Max intrinsic/preferred dimension is the dimension the widget could take, provided infinite
+    /// constraint on that axis.
+    ///
+    /// If axis == Axis::Horizontal, widget is being asked to calculate max intrinsic width.
+    /// If axis == Axis::Vertical, widget is being asked to calculate max intrinsic height.
+    ///
+    /// Box constraints must be honored in intrinsics computation.
+    ///
+    /// AspectRatioBox is an example where constraints are honored. If height is finite, max intrinsic
+    /// width is *height * ratio*.
+    /// Only when height is infinite, child's max intrinsic width is calculated.
+    ///
+    /// Intrinsic is a *could-be* value. It's the value a widget *could* have given infinite constraints.
+    /// This does not mean the value returned by layout() would be the same.
+    ///
+    /// Make sure this function doesn't return an infinite value.
```

Comment:
To be precise it's a *method* not just a *function*. We could say `This method **must** return a finite value.`